graphql/src/execution/query.rs

```diff
@@ -115,6 +152,24 @@ impl Query {
         max_complexity: Option<u64>,
         max_depth: u8,
     ) -> Result<Arc<Self>, Vec<QueryExecutionError>> {
+        let validation_errors = validate(
+            &schema.document(),
+            &query.document,
+            &GRAPHQL_VALIDATION_PLAN,
+        );
+
+        if validation_errors.len() > 0 {
+            return Err(validation_errors
+                .into_iter()
+                .map(|e| {
+                    QueryExecutionError::ValidationError(
+                        e.locations.clone().into_iter().nth(0),
```

Comment:
Perhaps here it's possible to just clone the first item instead of the whole `locations`.

Reply:
sure, fixed!